### ISSUE TITLE
Change `engine/testing/skia_gold_client` to the `flutter` environment

### ISSUE THREAD

**Issue by matanlurey** [flutter/flutter#157206]
There are [two hard-coded constants](https://github.com/flutter/engine/blob/08170c44b0f8932cadf8c7e01f139781c571530c/testing/skia_gold_client/lib/skia_gold_client.dart#L23-L24) that will need to be updated _or_ infer the correct environment.

One approach _could_ be:
```sh
flutter % git remote get-url origin
git@github.com:matanlurey/engine.git
```

If we check for `engine` it should cover cases such as `https://flutter.googlesource.com/mirrors/engine`.

**Comment by gaaclarke** [comment]
So the issue is that this dart code has hardcoded values in it, and when we do the monorepo we want to share this dart code between the framework and the engine, so the hardcoded values are no good?  I mean the easiest thing is to not share the code.  If we share the code people should specify what endpoint they want when they create a SkiaGoldClient.

**Comment by jtmcdole** [comment]
> so the hardcoded values are no good

Today they work great; but we're landing code in `flutter/flutter` to prep for the merge. After the merge, they'll be no good.

**Comment by matanlurey** [comment]
Right, it's not about sharing code or not (we should merge our X implementations of SkiaGoldClient once we merge repos, but it's orthogonal to not breaking the current implementation).

**Comment by gaaclarke** [comment]
I don't understand why the hardcoded values would not be good after the monorepo merge if we are fine with not sharing code?  They are a url and a repo name, they should be valid anywhere.

**Comment by matanlurey** [comment]
> I don't understand why the hardcoded values would not be good after the monorepo merge if we are fine with not sharing code? They are a url and a repo name, they should be valid anywhere.

I actually agree with @gaaclarke here - investing in an "auto-switching" mechanism for a switch that will happen exactly once seems foolish. The most I can suggest is adding a top-level constant flag that makes it easier for repo maintainers.

Wdut @jtmcdole?

**Comment by matanlurey** [comment]
Sg so I'll change this to "remember to change the hard-coded value".

**Comment by jtmcdole** [comment]
Will this blow up the merge somehow? I.e. we'll get the wrong goldens on push?

**Comment by matanlurey** [comment]
I am not sure exactly what will happen. I am guessing we are not going to turn on engine CI in the same PR we move everything over?

**Comment by matanlurey** [comment]
Working on this now.
